classes/announcement/Collector.inc.php

```diff
@@ -0,0 +1,134 @@
+<?php
+/**
+ * @file classes/announcement/Collector.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class announcement
+ *
+ * @brief A helper class to configure a Query Builder to get a collection of announcements
+ */
+
+namespace PKP\announcement;
+
+use HookRegistry;
+use Illuminate\Support\Facades\DB;
+use PKP\core\interfaces\CollectorInterface;
```

Comment:
FYI, the order of imports I'm following (not 100% consistent throughout code) is...

```
use PKP\something\somethingElse;
use PKP\another\thing;

use APP\foo\bar;
use APP\foo\baz;

use third\party\import;
use another\third\party;
```
I'm open to other options, but would like to standardize on a convention; this is just what I find most readable.
(I'm avoiding fancy syntax like `use some\namespace\{ClassA, ClassB, ClassC as C};` for now as I may need to batch-rewrite the imports using `sed` before this is all over.)

Reply:
I've been using my editor's automated code completion to add `use` statements. It puts them in alphabetical order. It looks like there is a code formatting rule that can do this for us: https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/3.0/doc/rules/class_notation/ordered_traits.rst

What do you think about using that?

Reply:
Automatic formatting of pretty much any kind would be great! But that option is for traits, not namespaces; I don't think there is an equivalent for namespaces, unfortunately.

Reply:
Ah you're right. I think this the one for imports: https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/3.0/doc/rules/import/ordered_imports.rst. Looks like there are a few options. Any preferences?

Reply:
I think just alphabetical. I'd prefer groupings separated by line breaks (`PKP\` and `APP\`) but it's a slight preference and not worth bikeshedding :)